test/extended/operators/certs.go

```diff
@@ -179,3 +200,153 @@ var _ = g.Describe("[sig-arch][Late]", func() {
+	serviceAccountObj := resourceread.ReadServiceAccountV1OrDie(serviceAccountYaml)
+	serviceAccountObj.Namespace = namespace
+	client := kubeClient.CoreV1().ServiceAccounts(namespace)
+	_, err := client.Create(ctx, serviceAccountObj, metav1.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return fmt.Errorf("error creating service account: %v", err)
+	}
+	return nil
+}
+
+func createRBACBindings(ctx context.Context, kubeClient kubernetes.Interface, namespace string) error {
+	roleBindingObj := resourceread.ReadRoleBindingV1OrDie(roleBindingPrivilegedYaml)
+	roleBindingObj.Namespace = namespace
+
+	client := kubeClient.RbacV1().RoleBindings(namespace)
+	_, err := client.Create(ctx, roleBindingObj, metav1.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return fmt.Errorf("error creating hostaccess SCC CRB: %v", err)
+	}
+	return nil
+}
+
+func createPods(ctx context.Context, kubeClient kubernetes.Interface, namespace string, nodeList *corev1.NodeList, testImagePullSpec string) error {
+	client := kubeClient.CoreV1().Pods(namespace)
+
+	podTemplate := resourceread.ReadPodV1OrDie(podYaml)
+	for _, node := range nodeList.Items {
+		podObj := podTemplate.DeepCopy()
+		podObj.Namespace = namespace
+		podObj.Spec.NodeName = node.Name
+		podObj.Spec.Containers[0].Image = testImagePullSpec
+
+		actualPod, err := client.Create(ctx, podObj, metav1.CreateOptions{})
+		if err != nil && !apierrors.IsAlreadyExists(err) {
+			return fmt.Errorf("error creating pod on node %s: %v", node.Name, err)
+		}
+
+		timeLimitedCtx, cancel := context.WithTimeout(ctx, time.Minute)
+		defer cancel()
+
+		if _, watchErr := watchtools.UntilWithSync(timeLimitedCtx,
+			cache.NewListWatchFromClient(
+				kubeClient.CoreV1().RESTClient(), "pods", namespace, fields.OneTermEqualSelector("metadata.name", actualPod.Name)),
+			&corev1.Pod{},
+			nil,
+			func(event watch.Event) (bool, error) {
+				pod := event.Object.(*corev1.Pod)
+				return pod.Status.Phase == corev1.PodSucceeded, nil
+			},
+		); watchErr != nil {
+			return fmt.Errorf("pod %s in namespace %s didn't complete: %v", actualPod.Name, namespace, watchErr)
+		}
+	}
+	return nil
+}
+
+func fetchNodePKIList(ctx context.Context, kubeClient kubernetes.Interface, node *corev1.Node) (*certgraphapi.PKIList, error) {
+	pkiList := &certgraphapi.PKIList{}
+
+	allBytes, err := nodeaccess.GetNodeLogFile(ctx, kubeClient, node.Name, certInspectResultFile)
+	if err != nil {
+		return pkiList, fmt.Errorf("failed to fetch file %s on node %s: %v", certInspectResultFile, node.Name, err)
+	}
+	err = json.Unmarshal(allBytes, pkiList)
+	if err != nil {
+		return pkiList, fmt.Errorf("failed to unmarshal file %s on node %s: %v", certInspectResultFile, node.Name, err)
+	}
+
+	return pkiList, nil
+}
+
+// GetOpenshiftTestsImagePullSpec returns the pull spec or an error.
+func getOpenshiftTestsImagePullSpec(ctx context.Context, adminRESTConfig *rest.Config) (string, error) {
```

Comment:
tbh, I don't remember how we did this for the networking check's monitortest, but let's expose that somewhere for internal consistency.

Reply:
Updated to reuse the same function